### ISSUE TITLE
A/B with Generic Stride's Failing GEMMSUP

### ISSUE THREAD

**Issue by xrq-phys** [flame/blis#484]
Hi.

I've been getting this kind of errors for Haswell builds:
```
% blis_<dt><op>_<params>_<stor>      m     n     k   gflops   resid      result
blis_dgemm_nn_ggg                  200   200     8   138.68   1.10e-02   FAILURE
blis_dgemm_nn_ggg                  225   225     8   142.78   1.19e-02   FAILURE
blis_dgemm_nn_ggg                  250   250     8   230.73   3.51e-03   FAILURE
blis_dgemm_nn_ggg                  275   275     8   299.95   4.34e-03   FAILURE
blis_dgemm_nn_ggg                  300   300     8   252.37   4.96e-03   FAILURE
```

It seems that `gg*` matrix multiplication is triggering `gemmsup` and returning `BLIS_SUCCESS` according to GDB:
```
% blis_<dt><op>_<params>_<stor>      m     n     k   gflops   resid      result

Breakpoint 1, bli_gemmsup (alpha=0x7fffffffa4c0, a=0x7fffffffa560, b=0x7fffffffa600, beta=0x7fffffffa6a0, c=0x7fffffffa740, cntx=0x0, rntm=0x0)
    at ../frame/3/bli_l3_sup.c:54
54		if ( bli_obj_dt( c ) != bli_obj_dt( a ) ||
(gdb) finish
Run till exit from #0  bli_gemmsup (alpha=0x7fffffffa4c0, a=0x7fffffffa560, b=0x7fffffffa600, beta=0x7fffffffa6a0, c=0x7fffffffa740, cntx=0x0, rntm=0x0)
    at ../frame/3/bli_l3_sup.c:54
0x0000000000576288 in bli_gemm (alpha=0x7fffffffa4c0, a=0x7fffffffa560, b=0x7fffffffa600, beta=0x7fffffffa6a0, c=0x7fffffffa740)
    at ../frame/3/bli_l3_oapi.c:104
104	GENFRONT( gemm )
Value returned is $1 = BLIS_SUCCESS
```

But according to my understanding of `gemmsup`, it should not be used (and return `BLIS_FAILURE`) if storage of B is general-strided because its kernel takes assumption `cs_b==1`.

I suppose 1 more check is needed in function `bli_gemmsup`?

**Comment by xrq-phys** [comment]
The same test failure is observed if I add GEMMSUP kernels for another march (A64fx).

**Comment by xrq-phys** [comment]
Adding this to `frame/3/bli_l3_sup.c` fixes the problem but it's apparently overkill: `gemmsup` only requires one side of A/B to be row or column major.

```
// Block general strided matrices.
if ( bli_obj_row_stride( a ) != 1 && bli_obj_col_stride( a ) != 1 ) return BLIS_FAILURE;
if ( bli_obj_row_stride( b ) != 1 && bli_obj_col_stride( b ) != 1 ) return BLIS_FAILURE;
```

**Comment by fgvanzee** [comment]
@xrq-phys Thanks for this report.

It seems the root problem is as follows: while BLIS's sup infrastructure currently attempts to catch cases that involve general stride and reject them with `BLIS_FAILURE` (in `frame/3/bli_l3_sup_int.c`, line 83), it does so from *within* the multithreaded region, and currently there is no attempt to sort through the various threads' return values and use that information to then return a single value back to `bli_gemmsup_ref()` and ultimately `bli_gemmsup()`. So it looks like I'll need to move that code from `bli_l3_sup_int.c` to somewhere outside of the threaded region.

There may have been a reason why I placed it there initially, but I can't seem to discern any reason why it needs to stay there. The only thing I can see is, that line 54 of `bli_l3_sup_int.c` is the first time we need to compute the `stor3_t` id for the current problem, so that may be why I waited until then. (And initially, this code was all single-threaded.)

I'll fix this later today!
